internal/pkg/skuba/metricsserver/metricsserver.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2020 SUSE LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package metricsserver
+
+import (
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	"k8s.io/kubernetes/cmd/kubeadm/app/util/pkiutil"
+
+	"github.com/SUSE/skuba/internal/pkg/skuba/kubernetes"
+	"github.com/SUSE/skuba/internal/pkg/skuba/util"
+)
+
+const (
+	certCommonName = "metrics-server.kube-system.svc" // DO NOT CHANGE THE CN
+	secretCertName = "metrics-server-cert"
+)
+
+// CreateCert creates a signed certificate for metrics-server
+// with kubernetes CA certificate and key
+func CreateCert(client clientset.Interface, pkiPath string) error {
```

Comment:
These two functions seem able to reuse in somewhere except constants. What do u think to move to a common place?

Reply:
I'd like to refactor it when doing certificate management feature. WDYT?

Reply:
Ha sure ;)